src/Utilities/universalfallback.jl

```diff
@@ -173,41 +173,76 @@ end
 # Name
 # The names of constraints not supported by `uf.model` need to be handled
 function MOI.set(uf::UniversalFallback, attr::MOI.ConstraintName, ci::CI{F, S}, name::String) where {F, S}
-    # TODO: Switch to lazily building the name-to-con dict like Model does.
-    if check_can_assign_name(uf, CI, ci, name)
-        if MOI.supports_constraint(uf.model, F, S)
-            MOI.set(uf.model, attr, ci, name)
-        else
-            setname(uf.connames, uf.namescon, ci, name)
-        end
+    if MOI.supports_constraint(uf.model, F, S)
+        MOI.set(uf.model, attr, ci, name)
+    else
+        uf.con_to_name[ci] = name
+        uf.name_to_con = nothing # Invalidate the name map.
     end
+    return
 end
 function MOI.get(uf::UniversalFallback, attr::MOI.ConstraintName, ci::CI{F, S}) where {F, S}
     if MOI.supports_constraint(uf.model, F, S)
         return MOI.get(uf.model, attr, ci)
     else
-        return get(uf.connames, ci, EMPTYSTRING)
+        return get(uf.con_to_name, ci, EMPTYSTRING)
+    end
+end
+
+function build_name_to_con_map(uf::UniversalFallback)
```

Comment:
Couldn't this function be shared with `MOIU.AbstractModel` ?

Reply:
Sharing internal functions across files without a proper interface (the function depends on an object having certain fields) makes the code harder to understand IMO. You shouldn't need to understand or think about implementation details of `Model` to understand `UniversalFallback`.

Reply:
You can slightly change the function, i.e. pass the fields instead of the model

Reply:
With [light bridges](https://github.com/JuliaOpt/MathOptInterface.jl/issues/523), bridges will not have an internal MOIU model but they will still need to handle names.
So this handling of names will be in UniversalFallback, AbstractModel and AbstractBridgeOptimizer.
If we choose to do the same thing in all case, having a separate utilities might help, it will also show that the name treatment done in each case is generic and not specific to the model.

Reply:
Done